### DISCUSSION TITLE
Trouble with elevation plot for final

### DISCUSSION THREAD

**Discussion by Adriannaluger** [earthlab-education/Earth-Analytics-2023-01-Intro#668] [! GitHub help]
Hi! I think I have successfully downloaded the elevations data, and done the steps prior to plotting by following the ndvi modis assignment. However, I keep getting this message when I attempt to plot: AttributeError: 'list' object has no attribute 'plot'. 

This is my sequence of code leading up to the plot. Let me know what I am missing... @eculler @nquarder 

`thunder_basin_gdf = (
    grassland_gdf
    .set_index('GRASSLANDN')
    .loc[['Thunder Basin National Grassland']]
)
download_key='Thunder-Basin-SRTM'
srtm_downloader = eaapp.AppeearsDownloader(
    download_key=download_key,
    product='SRTMGL1_NC.003',
    layer='SRTMGL1_DEM',
    start_date='02-11-2000',
    end_date='02-21-2000',
    polygon=thunder_basin_gdf)
srtm_downloader.download_files()`

`srtm_path = glob(
    os.path.join(
        srtm_downloader.data_dir, 
        'SRTMGL1_NC.003*',
        '*.tif')
)
[rxr.open_rasterio(srtm_path, masked=True).squeeze() for srtm_path in srtm_path][0]`

`doy_start = -19
doy_end = -12
scale_factor = 10000
srtm_da_list = []
for srtm_path in srtm_path:
    # Get date from file name
    doy = srtm_path[doy_start:doy_end]
    date = pd.to_datetime(doy, format='%Y%j')
    # Open dataset
    da = rxr.open_rasterio(srtm_path, masked=True).squeeze()
    # Prepare to concatenate: Add date dimension and clean up metadata
    da = da.assign_coords({'date': date})
    da = da.expand_dims({'date': 1})
    da.name = 'SRTM'
    # Divide by scale factor
    da = da / scale_factor
    # Add the DataArray to the end of the accumulator list
    srtm_da_list.append(da)
srtm_da_list`

`fig, ax = plt.subplots(figsize=(8,6))
srtm_da_list.plot(cmap=plt.cm.PiYG,robust=True,ax=ax)
Thunder_gdf_4326.boundary.plot(ax=ax,color='black')
plt.show()`

**Comment by eculler** [comment]
@Adriannaluger nice work!

What’s going on here is that you have a list of data arrays instead of one data array. So, you will need to get a single data array.

You could start by taking a look at the first one in the list (possibly the only one?). You can get it by indexing: srtm_da_list[0].plot(). If that works, great!

Otherwise, like if that one doesn’t cover the whole area, you could concat along the time dimension, and then perhaps take the mean along the time dimension. Check out the NDVI assignment where you worked with a time series for some examples.

**Comment by Adriannaluger** [comment]
@eculler I think that did it! Does this look okay to you?
<img width="349" alt="SRTM" src="https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/143222776/ac7a726a-2396-476f-b5eb-bb4f70f2d34d">

**Reply by eculler** [reply]
Looks good EXCEPT the SRTM units are off. Looks like you added a scale factor of 10000, but this dataset has a scale factor of 1.
